pkg/handlers/oauth.go

```diff
@@ -83,18 +83,26 @@ func NewOAuthHandler(authConfig *config.ServiceAuthConfig) *OAuthHandler {
 		}
 	}
 
-	hmacSecret := ""
-	if len(authConfig.HMACSecrets) > 0 {
-		hmacSecret = authConfig.HMACSecrets[0]
+	hmacSigningSecret := []byte{}
+	for i, hmacSecret := range authConfig.HMACSecrets {
+		decodedSecret, err := jwtauth.DecodeConfigValue(hmacSecret)
+		if err != nil {
+			log.Error().Err(err).Msgf("error decoding HMAC secret")
+			return nil
+		}
+		if i == 0 {
+			// The first secret is used to sign tokens - the rest are only used for validation (see middleware/auth.go)
```

Comment:
do we want to really exit when we have one secret that cannot be decoded and many more that can be decoded? 
Can we flag (delete) bad secrets and see what we have left and then assign hmacSigningSecret to the first good secret ?

Reply:
I changed my mind since I implemented the "skipping invalid" pattern above for client secrets. I think the user wants to know ASAP if something is invalid. If we start and skip invalid with error log, it is easier for them to miss it and would be frustrating to find later.

Reply:
If we are all in agreement, I should definitely change it to be the same above for client secrets.

Reply:
Created follow-up task: https://optimizely.atlassian.net/browse/OASIS-6108